src/compas_model/model/interactiongraph.py

```diff
@@ -84,8 +91,16 @@ def edge_interaction(self, edge):
         """
         return self.edge_attribute(edge, "interaction")  # type: ignore
 
-    def print_interactions(self):
-        """Print the interactions contained in the graph."""
+    def interactions_str(self):
```

Comment:
in my opinion this long string should be built by `__str__` itself, or by some kind of private method, but not by a public API method, since it is basically a utility of `__str__`...